src/Functions/URL/removeGarbageParametersFromURL.cpp

```diff
@@ -0,0 +1,427 @@
+            {
+                correction = -39;
+            }
+            *write = *i + correction;
+            ++write;
+        }
+
+        bool ret = pearsonIsUniform(n_c_start, n_c_end, '0', '?');
+        free(n_c_start);
+        return !ret;
+    }
+
+    /// @brief Check if string is long enough and if it could be hex or base64 for url
+    /// @param c_start Pointer to the first symbol of the string
+    /// @param c_end Pointer to the symbol following the last symbol of the string
+    /// @param min_len minimum string length
+    /// @return if the parameter should be included in URL
+    static bool process_content(const char * c_start, const char * c_end, uint64_t min_len)
+    {
+        if (static_cast<uint64_t>(c_end - c_start) < min_len)
+        {
+            return true;
+        }
+
+        bool check_hex = true;
+        bool check_base64 = true;
+        for (const char * i = c_start; i < c_end; ++i)
+        {
+            if (check_hex && !(('0' <= *i && *i <= '9') || ('a' <= *i && *i <= 'f')))
+            {
+                check_hex = false;
+            }
+            if (check_base64
+                && !(
+                    ('A' <= *i && *i <= 'Z') || ('a' <= *i && *i <= 'z') || ('0' <= *i && *i <= '9') || (*i == '-') || (*i == '_')
+                    || (*i == '=')))
+            {
+                check_base64 = false;
+                break;
+            }
+        }
+        if (!check_hex && !check_base64)
+        {
+            return true;
+        }
+        else if (check_hex && check_base64)
+        {
+            return checkHex(c_start, c_end) && checkBase64(c_start, c_end);
+        }
+        else
+        {
+            return checkBase64(c_start, c_end);
+        }
+    }
+
+    /// @brief Select the value of parameter and check if it is randomly generated
+    /// @param p_start Pointer to the first symbol of the parameter
+    /// @param p_end Pointer to the symbol following the last symbol of the parameter
+    /// @param min_len minimum length of parameter's value
+    /// @return if the parameter should be included in URL
+    static bool process_param(const char * p_start, const char * p_end, uint64_t min_len)
+    {
+        const char * c_start = nullptr;
+        for (const char * i = p_start; i < p_end; ++i)
+        {
+            if (*i == '=')
+            {
+                c_start = i + 1;
+                break;
+            }
+        }
+        if (c_start == nullptr)
+        {
+            return true;
+        }
+        return process_content(c_start, p_end, min_len);
+    }
+
+    /// @brief Add data to string
+    /// @param res_data String to add data to
+    /// @param size length of data to be added
+    /// @param pdata pointer to the first character of data
+    /// @param res_offsets offsets of output string
+    static void copy(ColumnString::Chars & res_data, size_t size, const char * pdata, ColumnString::Offsets & res_offsets)
```

Comment:
`copyIntoOutput`.

Also let's bring the parameters into a more logical order: `const char * from, size_t from_length, ColumnString::Chars & res_data, ColumnString::Offsets & res_offsets`